web/src/components/dashboard/BusinessStructurePrompt.test.tsx

```diff
@@ -0,0 +1,52 @@
+import { BusinessStructurePrompt } from "@/components/dashboard/BusinessStructurePrompt";
+import { getMergedConfig } from "@/contexts/configContext";
+import { ROUTES } from "@/lib/domain-logic/routes";
+import * as mockRouter from "@/test/mock/mockRouter";
+import { useMockRouter } from "@/test/mock/mockRouter";
+import { fireEvent, render, screen } from "@testing-library/react";
+
+jest.mock("@/lib/data-hooks/useRoadmap", () => ({ useRoadmap: jest.fn() }));
+jest.mock("next/router", () => ({ useRouter: jest.fn() }));
+
+describe("<BusinessStructurePrompt />", () => {
+  const Config = getMergedConfig();
+
+  beforeEach(() => {
+    jest.resetAllMocks();
+    useMockRouter({});
+  });
+
+  it("hides business structure prompt button", () => {
+    render(<BusinessStructurePrompt isButtonHidden={true} />);
+    expect(screen.queryByTestId("business-structure-prompt-button")).not.toBeInTheDocument();
+  });
+
+  it("shows business structure prompt button", () => {
+    render(<BusinessStructurePrompt />);
+    expect(screen.getByTestId("business-structure-prompt-button")).toBeInTheDocument();
+  });
+
+  it("renders notCompletedTaskPromptAnyTask from config", () => {
+    useMockRouter({
+      asPath: "any",
+    });
+    render(<BusinessStructurePrompt />);
+    expect(screen.getByTestId("any-task-content")).toBeInTheDocument();
+    expect(screen.queryByTestId("business-structure-task-content")).not.toBeInTheDocument();
+  });
+
+  it("renders notCompletedTaskPromptBusinessStructureTask from config", () => {
```

Comment:
it would be helpful for these test names to include the "when" because otherwise they read as conflicting to the previous statement

"hides business structure prompt button when isButtonHidden is true"
"hides business structure prompt button when isButtonHidden is false or not included"
"renders generic config when on any task"
"renders specialized config when on business structure task"